File: sdks/dotnet/src/Jobcelis/JobcelisException.cs
namespace Jobcelis;

/// <summary>
/// Exception thrown when the Jobcelis API returns an error.
/// </summary>
public class JobcelisException : Exception
{
    /// <summary>HTTP status code.</summary>
    public int StatusCode { get; }

    /// <summary>Error detail from the API response.</summary>
    public object? Detail { get; }

    public JobcelisException(int statusCode, object? detail)
        : base($"HTTP {statusCode}: {detail}")
    {
        StatusCode = statusCode;
        Detail = detail;
    }
}

File: sdks/php/src/JobcelisException.php
<?php

declare(strict_types=1);

namespace Jobcelis;

/**
 * Exception thrown when the Jobcelis API returns an error.
 */
class JobcelisException extends \Exception
{
    public readonly int $statusCode;
    public readonly mixed $detail;

    public function __construct(int $statusCode, mixed $detail)
    {
        $this->statusCode = $statusCode;
        $this->detail = $detail;

        $message = is_string($detail)
            ? "HTTP {$statusCode}: {$detail}"
            : "HTTP {$statusCode}: " . json_encode($detail);

        parent::__construct($message, $statusCode);
    }
}
